api/config/routes.rb

```diff
@@ -133,6 +133,16 @@
       end
     end
 
+    resources :address_book, only: [], param: :address_id do
+      collection do
+        get :mine
+        post :update
+      end
+      member do
+        post :remove
+      end
+    end
```

Comment:
As I understand relation correctly, then we should have `resources :addresses` that is automativally in User scope. User can manage his addresses.